src/util/name_generator.ts

```diff
@@ -0,0 +1,39 @@
+export class NameGenerator {
+    private counters: { [base: string]: number };
```

Comment:
We can use the object here, ofcourse. But would be cool to have a `Map<string, number>` instead. It may have some builerplate code, but on large amount of keys it would be more efficient.